js/modules/k6/experimental/csv/module.go

```diff
@@ -0,0 +1,238 @@
+func (mi *ModuleInstance) NewParser(call goja.ConstructorCall) *goja.Object {
+	rt := mi.vu.Runtime()
+
+	if len(call.Arguments) < 1 || goja.IsUndefined(call.Argument(0)) {
+		common.Throw(rt, fmt.Errorf("parser constructor takes at least one non-nil source argument"))
+	}
+
+	// 1. Make sure the Goja object is a fs.File (goja operation)
+	var file fs.File
+	if err := mi.vu.Runtime().ExportTo(call.Argument(0), &file); err != nil {
+		common.Throw(mi.vu.Runtime(), fmt.Errorf("first arg doesn't appear to be a *file.File instance"))
+	}
+
+	var options parserOptions
+	if len(call.Arguments) == 2 && !goja.IsUndefined(call.Argument(1)) {
+		var err error
+		options, err = newParserOptionsFrom(call.Argument(1).ToObject(rt))
+		if err != nil {
+			common.Throw(rt, fmt.Errorf("encountered an error while interpreting Parser options; reason: %w", err))
+		}
+	}
+
+	// Instantiate and configure csv reader
+	r := csv.NewReader(&file.Impl)
+	r.ReuseRecord = true        // evaluate if this is needed, and if it leads to unforeseen issues
+	r.Comma = options.Delimiter // default delimiter, should be modifiable by the user
+
+	var (
+		fromLineSet      = options.FromLine.Valid
+		skipFirstLineSet = options.SkipFirstLine
+	)
+
+	// If the user wants to skip the first line, we consume and discard it.
+	if skipFirstLineSet && (!fromLineSet || options.FromLine.Int64 == 0) {
+		_, err := r.Read()
+		if err != nil {
+			common.Throw(rt, fmt.Errorf("failed to skip the first line; reason: %w", err))
+		}
+	}
+
+	// If the user wants to start reading from a specific line, we read and discard
+	// lines until we reach the desired line.
+	if fromLineSet && options.FromLine.Int64 > 0 {
+		for i := int64(0); i < options.FromLine.Int64; i++ {
+			_, err := r.Read()
+			if err != nil {
+				common.Throw(rt, fmt.Errorf("failed to iterate to fromLine; reason: %w", err))
+			}
+		}
+	}
+
+	// Create a new Parser instance
+	parser := Parser{
+		reader:  r,
+		options: options,
+		vu:      mi.vu,
+	}
+
+	return rt.ToValue(&parser).ToObject(rt)
+}
+
+// Next returns the next row in the CSV file.
+func (p *Parser) Next() *goja.Promise {
+	promise, resolve, reject := promises.New(p.vu)
+
+	go func() {
+		var records []string
+		var done bool
+		var err error
+
+		if p.options.ToLine.Valid && p.CurrentLine.Load() >= p.options.ToLine.Int64 {
+			done = true
+			resolve(parseResult{done, records})
+			return
+		}
+
+		records, err = p.reader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				done = true
+			} else {
+				reject(err)
+				return
+			}
+		}
+
+		p.CurrentLine.Add(1)
+
+		resolve(parseResult{done, records})
+	}()
+
+	return promise
+}
+
+// parseResult holds the result of a CSV parser's parsing operation such
+// as when calling the [Next].
+type parseResult struct {
+	// Done indicates whether the parser has finished reading the file.
+	Done bool `js:"done"`
+
+	// Value holds the line's records value.
+	Value []string `js:"value"`
```

Comment:
Nit; why not using the plural form, as I guess it will generally contain multiple values, thus it'd be more clear that it's an array, especially considering how little explicit is JS.

Reply:
Agreed! That would be nice 👍🏻